sqlmesh/core/context_diff.py

```diff
@@ -118,7 +121,7 @@ def create(
         env = state_reader.get_environment(environment)
 
         create_from_env_exists = False
-        if env is None or env.expired:
+        if env is None or env.expired or env.gateway_managed != gateway_managed_virtual_layer:
```

Comment:
Shouldn't we also reflect this change in https://github.com/TobikoData/sqlmesh/blob/b31c898d6834cc8147ec8a506ec2dc563ae2a733/sqlmesh/core/context_diff.py#L273? 

Otherwise, if the flag is the only thing that changed the planner won't detect a difference.